karmagrambot/commands.py

```diff
@@ -66,7 +77,14 @@ def karmas(_: Bot, update: Update):
         bot: The object that represents the Telegram Bot.
         update: The object that represents an incoming update for the bot to handle.
     """
-    top_users = analytics.get_top_n_karmas(update.message.chat.id, 10)
+    text = update.message.text
+    _, *args = text.split()
+    arg = args[0].lstrip('-') if args else None
+    period = get_period('m')
```

Comment:
I would prefer something like:

```python
_, *args = text.split()
arg, = args if args else ('m',)
requested_period = arg.lstrip('-')

if requested_period not in ('m', 'month', 'w', 'week, 'y', 'year', 'all', 'alltime'):
    update.message.reply(f"Period {requested_period} is not supported.")
    return

period = get_period(requested_period)
...
```

Reply:
What wizardry is that?
`arg, = args if args else ('m',)` is the same as `arg = args[0] if args else 'm'` ?
I personally think the second option is more clear.

The rest of the suggestion I updated.

Reply:
That's fine.